app/src/app/[lng]/data/review/page.tsx

```diff
@@ -0,0 +1,565 @@
+"use client";
+
+import FileDataCard from "@/components/Cards/file-data-card";
+
+import { BuildingIcon } from "@/components/icons";
+import Wrapper from "@/components/wrapper";
+import { useTranslation } from "@/i18n/client";
+import { RootState } from "@/lib/store";
+
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import {
+  Box,
+  Button,
+  Card,
+  Heading,
+  Icon,
+  IconButton,
+  Text,
+} from "@chakra-ui/react";
+import { useRouter } from "next/navigation";
+
+import { FaRegTrashAlt, FaTrash } from "react-icons/fa";
+import { FiTrash, FiTrash2 } from "react-icons/fi";
+import { MdOutlineEdit } from "react-icons/md";
+import { useSelector, useDispatch } from "react-redux";
+import { clear, removeSectorData } from "@/features/city/inventoryDataSlice";
+import { api } from "@/services/api";
+import { appendFileToFormData } from "@/util/helpers";
+
+export default function ReviewPage({
+  params: { lng },
+}: {
+  params: { lng: string };
+}) {
+  const { t } = useTranslation(lng, "data");
+  const router = useRouter();
+  const dispatch = useDispatch();
+  const getAllSectorData = useSelector(
+    (state: RootState) => state.inventoryData.sectors,
+  );
+  const onBack = () => {
+    router.push("/data/3");
+  };
+  const onDiscard = () => {
+    dispatch(clear());
+    // router.push("/data");
+  };
+
+  const stationaryEnergy = getAllSectorData.filter(
+    (sector) => sector.sectorName === "Stationary Energy",
+  );
+  const transportation = getAllSectorData.filter(
+    (sector) => sector.sectorName === "Transportation",
+  );
+  const waterAndWasteWater = getAllSectorData.filter(
+    (sector) => sector.sectorName === "Waste and wastewater",
+  );
+
+  const onDiscardSectorChanges = (sector: string) => {
+    dispatch(removeSectorData({ sectorName: sector }));
+  };
+
+  const [addUserFile] = api.useAddUserFileMutation();
+
+  const onConfirm = async () => {
+    for (const sector of getAllSectorData) {
+      const formData = new FormData();
+      for (const fileData of sector.files) {
+        const file = appendFileToFormData(
+          fileData.data,
+          `file_XXX.${fileData.fileType}`,
+        );
+
+        formData.append("userId", fileData.userId!);
+        formData.append("sector", fileData.sector!);
+        formData.append("status", fileData.status!);
+        formData.append("fileReference", "");
+        formData.append("url", fileData.url!);
+        formData.append("gpcRefNo", "");
+        formData.append("data", file, file.name);
+      }
+
+      await addUserFile(formData).then(() => {
+        // TODO
+        // Trigger notification to user
+        dispatch(clear());
+        router.push("/");
```

Comment:
This is weird to me. There can be multiple files, so either await Promise.all or do the router/ clear logic somewhere else.